src/services/execution-api/service.ts

```diff
@@ -112,25 +200,154 @@ export const makeExecutionApi = (
 
     const { result } = logsDTO(json)
 
-    logger.info(`Loaded ${result.length} events`)
+    logger.info('Loaded ValidatorExitRequest events', { amount: result.length })
+
+    const validatorsToEject: {
+      validatorIndex: string
+      validatorPubkey: string
+    }[] = []
+
+    for (const log of result) {
+      const parsedLog = iface.parseLog(log)
 
-    const valsToEject = result.map((log) => {
-      const parsed = iface.parseLog(log)
-      return {
-        validatorIndex: parsed.args['validatorIndex'] as string,
-        validatorPubkey: parsed.args['validatorPubkey'] as string,
+      const { validatorIndex, validatorPubkey } = parsedLog.args as unknown as {
+        validatorIndex: ethers.BigNumber
+        validatorPubkey: string
       }
-    })
 
-    return valsToEject
+      try {
+        await verifyEvent(
+          validatorPubkey,
+          log.transactionHash,
+          parseInt(log.blockNumber)
+        )
+        logger.debug('Event security check passed', { validatorPubkey })
+      } catch (e) {
+        logger.error(`Event security check failed for ${validatorPubkey}`, e)
+        continue
+      }
+
+      validatorsToEject.push({
+        validatorIndex: validatorIndex.toString(),
+        validatorPubkey,
+      })
+    }
+
+    return validatorsToEject
+  }
+
+  const verifyEvent = async (
+    validatorPubkey: string,
+    transactionHash: string,
+    toBlock: number
+  ) => {
+    // Final tx in which report data has been finalized
+    const finalizationTx = await getTransaction(transactionHash)
```

Comment:
maybe `finalization` is not a good naming. 
`submitReport` & `submitReportData` in contracts
reportHash & reportData?